docs/install-and-setup.md

```diff
@@ -257,22 +257,32 @@ source <(COMPLETE=zsh jj)
 
 ### Fish
 
-!!! note
-
-    No configuration is required with fish >= 4.1 which loads dynamic completions by default.
 
 #### Standard
 
 ```shell
+# Not recommended, use the dynamic completions instead
 jj util completion fish | source
 ```
 
 #### Dynamic
 
 ```shell
+# Only needed with fish < 4.1. fish >= 4.1 loads these by default
 COMPLETE=fish jj | source
 ```
 
+!!! note
+
+    If dynamic completion work only partially, or if you get only static completions,
+    this might be because your `fish` version is older than 4.1 or because the static
+    completions are loaded by your personal config and/or by your distribution's
+    config. In the latter case, contact whomever made the `jj` package you are using.
```

Comment:
may be worth noting that, as a workaround for this, one may drop `COMPLETE=fish jj | source` into `~/.config/fish/completions` -- that will override any `jj.fish` from old fish or a distro package.